Mathlib/Algebra/Group/AddChar.lean

```diff
@@ -61,22 +62,23 @@ namespace AddChar
 -- Porting note(https://github.com/leanprover-community/mathlib4/issues/5020): added
 section DerivedInstances
 
-variable (A : Type u) [AddMonoid A] (M : Type v) [CommMonoid M]
 
-instance : CommMonoid (AddChar A M) :=
+variable (A : Type u) (M : Type v)
+
+instance [AddZeroClass A] [CommMonoid M] : CommMonoid (AddChar A M) :=
   inferInstanceAs (CommMonoid (Multiplicative A →* M))
 
-instance : Inhabited (AddChar A M) :=
+instance [AddZeroClass A] [Monoid M] : Inhabited (AddChar A M) :=
```

Comment:
Is it really needed? I commented it and everything compiles.

Reply:
You mean the inhabited instance? We had a linter for it in Lean 3 to avoid writing void structures.

Reply:
It uses `inferInstanceAs`, which usually means that it is not found automatically.

Reply:
This is not new code in this PR (I just weakened the typeclass assumptions slightly) so I propose to leave it as it is for now.

Reply:
Yes it is out of scope